gufe/protocols/protocoldag.py

```diff
@@ -45,23 +163,70 @@ def name(self):
     def graph(self):
         return self._graph
 
-    def execute(self) -> Union[ProtocolDAGResult, ProtocolDAGFailure]:
-        """Execute the full DAG in-serial, in process."""
+    def protocol_units(self):
+        return list(self._protocol_units)
+
+    def execute(self, *, 
```

Comment:
Not sure I quite get the execute for a whole DAG, it's not really the recommended usage and misleading. I'd turn this into a convenience function rather than a method.

Reply:
Fair enough, will move it to a function so it's not part of the API of `ProtocolDAG`.